pykafka/protocol.py

```diff
@@ -191,16 +192,20 @@ def pack_into(self, buff, offset):
         if self.partition_key is None:
             fmt = '!BBii%ds' % len(self.value)
             args = (self.MAGIC, self.compression_type, -1,
-                    len(self.value), self.value)
+                    len(self.value), get_bytes(self.value))
         else:
             fmt = '!BBi%dsi%ds' % (len(self.partition_key), len(self.value))
             args = (self.MAGIC, self.compression_type,
-                    len(self.partition_key), self.partition_key,
-                    len(self.value), self.value)
+                    len(self.partition_key),
+                    get_bytes(self.partition_key),
+                    len(self.value),
+                    get_bytes(self.value))
+
         struct.pack_into(fmt, buff, offset + 4, *args)
         fmt_size = struct.calcsize(fmt)
-        crc = crc32(buffer(buff[(offset + 4):(offset + 4 + fmt_size)]))
-        struct.pack_into('!i', buff, offset, crc)
+        data = buffer(buff[(offset + 4):(offset + 4 + fmt_size)])
+        crc = crc32(data) & 0xffffffff
+        struct.pack_into('!I', buff, offset, crc)
```

Comment:
I changed it because the tests were larger than a signed int, you can see it doing something like:


```
>>> from binascii import crc32
>>> crc32(b'hello') & 0xffffffff
907060870
>>> 2147483647 > crc32(b'hello bar') & 0xffffffff
False
```

but even we were able to change it back we should have the backwards compatibility issue because the previous code wasn't doing the bit shifting so we'd have different results between python2 and python3:

Python2:

```
>>> crc32(b'hello bar')
-1737004441
>>> crc32(b'hello bar') & 0xffffffff
2557962855
```

Python3:

```
>>> from binascii import crc32
>>> crc32(b'hello bar')
2557962855
>>> crc32(b'hello bar') & 0xffffffff
2557962855
```